pkg/controller/controller_broker.go

```diff
@@ -363,14 +430,62 @@ func (c *controller) reconcileServiceClassFromServiceBrokerCatalog(broker *v1alp
 
 	toUpdate := clone.(*v1alpha1.ServiceClass)
 	toUpdate.Bindable = serviceClass.Bindable
-	toUpdate.Plans = serviceClass.Plans
 	toUpdate.PlanUpdatable = serviceClass.PlanUpdatable
 	toUpdate.Tags = serviceClass.Tags
 	toUpdate.Description = serviceClass.Description
 	toUpdate.Requires = serviceClass.Requires
 
 	if _, err := c.serviceCatalogClient.ServiceClasses().Update(toUpdate); err != nil {
-		glog.Errorf("Error updating serviceClass %v from ServiceBroker %v: %v", serviceClass.Name, broker.Name, err)
+		glog.Errorf("Error updating serviceClass %q from ServiceBroker %q: %v", serviceClass.Name, broker.Name, err)
+		return err
+	}
+
+	return nil
+}
+
+// reconcileServicePlan reconciles a ServicePlan after the
+// ServiceClass's catalog has been re-listed.
+func (c *controller) reconcileServicePlan(servicePlan *v1alpha1.ServicePlan) error {
+	existingServicePlan, err := c.servicePlanLister.Get(servicePlan.Name)
+	if errors.IsNotFound(err) {
+		// An error returned from a lister Get call means that the object does
+		// not exist.  Create a new ServicePlan.
+		if _, err := c.serviceCatalogClient.ServicePlans().Create(servicePlan); err != nil {
+			glog.Errorf("Error creating servicePlan %q: %v", servicePlan.Name, err)
+			return err
+		}
+
+		return nil
+	} else if err != nil {
+		glog.Errorf("Error getting servicePlan %q: %v", servicePlan.Name, err)
+		return err
+	}
+
+	if existingServicePlan.ExternalID != servicePlan.ExternalID {
+		errMsg := fmt.Sprintf("ServicePlan %q already exists with OSB guid %q, received different guid %q", servicePlan.Name, existingServicePlan.ExternalID, servicePlan.ExternalID)
+		glog.Error(errMsg)
+		return fmt.Errorf(errMsg)
+	}
+
+	glog.V(5).Infof("Found existing servicePlan %q; updating", servicePlan.Name)
+
+	// There was an existing service plan -- project the update onto it and
+	// update it.
+	clone, err := api.Scheme.DeepCopy(existingServicePlan)
```

Comment:
I think for the time being there should be a check here to see whether the external ID is changing

Reply:
Isn't this something the validation should do. I'll add it there.